### DISCUSSION TITLE
Chaining elements: Wdio custom command

### DISCUSSION THREAD

**Discussion by NavyMint** [webdriverio/codemod#44] [General]
Hello dear Wdio team :)
We still trying to migrate page objects from protractor to webdriverio and bumped into next problem - we can't use custom command chaining on Promise\< Element \>

**in Page Object:**

private parentElement = $('element')

get product(): CustomWebWrapper {
       return new CustomWebWrapper( this.parentElement.dataId('childElement'))
}

**in conf file:**

 browser.addCommand('dataId', function(selector: string) {
         return this.$$(`[data-id=${selector}]`);
 }, true);

The problem is to use dataId command in chain if previous element not awaited. I mean this way it will work :

async product(): Promise<CustomWebWrapper> {
       return new CustomWebWrapper((await this.parentElement).dataId('anyChildLocator'));
}

but then we need to rework all usages of getters to functions and also rewrite getters to async functions. Is there any workaround to use custom commands on Promise<Element> ?

**Comment by christian-bromann** [comment]
> Is there any workaround to use custom commands on Promise ?

We are in the process of improving the async API which should make your use case possible. See more here: https://github.com/webdriverio/webdriverio/blob/cb-async-api-blog-post/website/blog/2021-07-25-sync-api-deprecation.md. A release is expected within the next days.

**Reply by NavyMint** [reply]
thank you, Christian!! Its definitely our case, and this improvement will really save a lot of our time during migration,  then we will wait for a new version
